install/update_95_xx/reservationitem.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2020 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+function reservationItemAddUnicity(Migration $migration) {
+   global $DB;
+
+   $table = ReservationItem::getTable();
+
+   // Drop is_deleted
+   $migration->dropKey($table, 'is_deleted');
+   $migration->dropField($table, "is_deleted");
+
+   // Find duplicates by itemtype/items_id couple
+   $duplicates_by_itemtype = $DB->request([
+      'SELECT'  => ['itemtype', 'items_id'],
+      'COUNT'   => 'cpt',
+      'FROM'    => $table,
+      'WHERE'   => [],
+      'GROUPBY' => ['itemtype', 'items_id'],
+      'HAVING'  => ['cpt' => ['>' , 1]]
+   ]);
+
+   foreach ($duplicates_by_itemtype as $duplicates) {
+      // Get ids of all duplicate for this itemtype/items_id couple expect the first
+      // LIMIT 5000 is there because we need a LIMIT clause to use OFFSET
+      // We dont expect more than a couple duplicates so 5000 is fine
+      $to_delete = $DB->request([
+         'SELECT' =>  'id',
+         'FROM' => $table,
+         'WHERE' => [
+            'itemtype' => $duplicates['itemtype'],
+            'items_id' => $duplicates['items_id'],
+         ],
+         'LIMIT' => 5000,
```

Comment:
I guess you could use this instead of a limit pulled out of a hat:
```suggestion
         'LIMIT' => $duplicates['cpt'],
```